File: core/src/main/java/org/granchi/hollywood/ModelExceptionHandler.java
package org.granchi.hollywood;

/**
 * Handler for reacting to exceptions during Model.actUpon(Action).
 *
 * Without it, the application would end if an Exception happens.
 *
 * @author serandel
 */
public interface ModelExceptionHandler {
    /**
     * An Exception has ocurred during Model.actUpon(Action).
     *
     * @param model Model that received the Action
     * @param action Action that was acted upon
     * @param exception Exception thrown
     * @return Model that the application should keep as the current model, null for finishing the app
     */
    Model onException(Model model, Action action, Exception exception);
}

File: core/src/main/kotlin/org/granchi/hollywood/ModelExceptionHandler.kt
package org.granchi.hollywood

/**
 * Handler for reacting to exceptions during [Model.actUpon(Action)].
 *
 * Without it, the application would end if an Exception happens.
 */
interface ModelExceptionHandler {
    /**
     * Return the next [Model] to use in the application, or null to end it, when an Exception has
     * ocurred during [Model.actUpon(Action)].
     */
    fun onException(model: Model, action: Action, exception: Exception): Model?
}
